### PR TITLE
HW3 from Yanran Wei

### PR DESCRIPTION
**Pull request by Blairwei**

### CODE REVIEW

File: 03_good_programming_R_functions/HW3_Wei_Yanran.Rmd [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+colnames(result_q6) <- c("Observers", "mean1", "mean2", "sd1", "sd2","correlation")
+for(i in 1:13){
+  middle_q6<-subset(data_q6,data_q6[, 1]==i)
+  result_q6[i,1] = mean(middle_q6[, 1])
+  result_q6[i,2] = mean(middle_q6[, 2])
+  result_q6[i,3] = mean(middle_q6[, 3])
+  result_q6[i,4] = sd(middle_q6[, 2])
+  result_q6[i,5] = sd(middle_q6[, 3])
+  result_q6[i,6] = cor(middle_q6[, 2], middle_q6[, 3])
+}
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Wei_Yanran.Rmd]: function???

**Comment by Blairwei** [comment]
Hi,
I have added echo=false in the code chunk and made little adjustment to problem 6 to create a function.
Please advise.

**Comment by Blairwei** [comment]
Hi,
I have added echo=false in the code chunk and make little adjustments to problem 6 to create a function.
Please advise. Thank you.

**Comment by Blairwei** [comment]
Hi,
I have added echo=false in the code chunk and made little adjustments to problem 6 to create a function.
Please advise. Thank you.

File: 03_good_programming_R_functions/HW3_Wei_Yanran.Rmd [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+data_q6<-readRDS("C:/Users/Echo/Desktop/2017 Fall/Statistical Package/HW3/HW3_data.rds")
+result_q6 <- matrix(nrow = 13,ncol = 6)
+colnames(result_q6) <- c("Observers", "mean1", "mean2", "sd1", "sd2","correlation")
+stat <- function(x){
+  middle_q6<-subset(data_q6,data_q6[, 1]==x)
+  d <- c(mean(middle_q6[, 1]), mean(middle_q6[, 2]), mean(middle_q6[, 3]), sd(middle_q6[, 2]),   sd(middle_q6[, 3]), cor(middle_q6[, 2], middle_q6[, 3]))
+  return(d)
+}
+for(i in 1:13){
+  result_q6[i, ] = stat(i)
```
Comment (rsettlage) [inline 03_good_programming_R_functions/HW3_Wei_Yanran.Rmd]: certainly, it is now a function

next time, you might consider the good practices guidelines:
1. functions do things, name them with verbs, i.e. createSumStats or something 
2. functions have input and output, you should add that to a comment inside the function
3. EVERYthing you need to perform the function should be provided TO the function as arguments, data and all

**Pull request closed without merging**